MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/ViewModels/DetailTitleViewModel.swift

```diff
@@ -11,3 +11,19 @@ struct DetailTitleViewModel {
     let posterPath: URL?
     
 }
+
+extension DetailTitleViewModel {
+    
+    init(fromModel model: MovieModel) {
+        self.init(
```

Comment:
I wanted to use this code in `self.init()`, but the the app wouldn't build. It actually does the same thing, but this way the code looks nicer. 
```
        guard let date = model.releaseDate else { return }

        let dateSplitted = date.split(separator: "-")
        let dateString = "\(dateSplitted[1])/\(dateSplitted[2])/\(dateSplitted[0])"

        let hours = (model.runtime ?? 0) / 60
        let minutes = (model.runtime ?? 0) % 60
        
        self.init(
            title: model.title,
            year: String(dateSplitted[0]),
            releaseDate: dateString,
            genres: model.genreIds!.map { Genre(rawValue: $0)?.genreName ?? "" },
            duration: "\(hours)h \(minutes)m",
            userScore: Int(model.voteAverage * 10),
            posterPath: model.posterPath)
```

Reply:
The only problem I see with the code above is that you have `guard`.
In those cases, you need to use optional init:
```swift
init?(fromModel model: MovieModel) {
    guard let date = model.releaseDate else { return nil }

    // the rest of the code
}
```

Also, if `date` shouldn't block the initializer, you can try to unwrap it and if it fails, set the `dateString` with `""`.

Reply:
[Changed](https://github.com/lihter/Movie-App/pull/30/commits/cca37b805fab60d09dcbfc057d25f02fa279dbbc#diff-574b7dfe6f424da91711f2a6dda3c734b3b93745b9bf5ee3af14494f571f8f11R17)